python/mxnet/gluon/utils.py

```diff
@@ -115,8 +115,21 @@ def split_and_load(data, ctx_list, batch_axis=0, even_split=True):
     return [i.as_in_context(ctx) for i, ctx in zip(slices, ctx_list)]
 
 
-def clip_global_norm(arrays, max_norm):
+def clip_global_norm(arrays, max_norm, check_isfinite=True, check_scale=True):
     """Rescales NDArrays so that the sum of their 2-norm is smaller than `max_norm`.
+
+    Parameters
+    ----------
+    arrays : list of NDArray
+    max_norm : float
+    check_isfinite : bool, default True
+         If True, check that the total_norm is finite (not nan or inf). This
+         requires a blocking .asscalar() call.
+    check_scale : bool, default True
+         If True, skip array rescaling if max_norm / total_norm >= 1. This
+         requires a blocking call. If False, rescale arrays with min(1,
```

Comment:
rescale is not blocking

Reply:
Yes. But previously there is a blocking call to check if re-scale is necessary.
If `check_scale is False`, we always rescale, possibly by `1`.

Reply:
I see. Maybe we can use contrib.cond for this

Reply:
`contrib.cond` would only help when working with the symbolic interface. For ndarray, `contrib.cond` also uses a blocking `.asscalar()` call. `clip_global_norm` always works with the ndarray API.

Reply:
For now the PR preserves the old default behavior. I'm happy to remove the `check_scale` argument and to always rescale, trading off computation against avoiding blocking calls. That would assume that it is always cheaper/better do the rescaling than to wait for a blocking asscalar() and potentially avoid rescaling.

Reply:
+1 for removing check_scale. It doesn't seem necessary. We can always perform a scaling.